### ISSUE TITLE
Surprising behavior of underscore-prefixed records

### ISSUE THREAD

**Issue by emmanuel** [skynetservices/skydns#71]
I noticed that wildcard queries behave differently for underscore-prefixed records. Example:

``` bash
root@06d9a36fb32b:/# dig +short SRV _zookeeper._tcp.test.cluster.local
10 33 2181 zookeeper-1.test.cluster.local.
10 33 2181 zookeeper-2.test.cluster.local.
10 33 2181 zookeeper-3.test.cluster.local.
root@06d9a36fb32b:/# dig +short SRV _tcp.test.cluster.local
root@06d9a36fb32b:/# dig +short SRV zookeeper.tcp.test.cluster.local
10 33 2181 zookeeper-1.test.cluster.local.
10 33 2181 zookeeper-2.test.cluster.local.
10 33 2181 zookeeper-3.test.cluster.local.
root@06d9a36fb32b:/# dig +short SRV tcp.test.cluster.local
10 33 2181 zookeeper-1.test.cluster.local.
10 33 2181 zookeeper-2.test.cluster.local.
10 33 2181 zookeeper-3.test.cluster.local.
root@06d9a36fb32b:/# dig +short zookeeper-1.test.cluster.local
e11ccacf6d174348a1578f045886f982.hosts.cluster.local.
172.31.46.183
root@06d9a36fb32b:/# dig +short _zookeeper._tcp.test.cluster.local
root@06d9a36fb32b:/# dig +short 1._zookeeper._tcp.test.cluster.local
zookeeper-1.test.cluster.local.
e11ccacf6d174348a1578f045886f982.hosts.cluster.local.
172.31.46.183
```

Why the difference in behavior of `dig +short SRV _tcp.test.cluster.local` vs `dig +short SRV tcp.test.cluster.local`?

**Comment by miekg** [comment]
This should be fixed with the release of etcd 2.0?

**Comment by drewcsillag** [comment]
I left the company I was working with when I ran into this, so I haven't followed the etcd 2 release at all.
